### ISSUE TITLE
Speed up by only showing "definitions" + load only relevant tiddlers

### ISSUE THREAD

**Issue by twMat** [erwanm/tw-aggregator#76]
@danielo515 in his [TiddlyDrive](http://danielorodriguez.com/TiddlyDrive4Community/#How%20does%20this%20work%3F) (sadly and abandoned project because he got tired of Google changing API) explains (in that link) the workings for TiddlyDrive:

>The tiddlers built by the community are not directly displayed or downloaded. Instead, a list of available tiddlers is downloaded on startup (probably you noticed that message in the top right corner). Those tiddlers are just definitions of the real tiddlers. If you want to see that tiddler just click the download button and you will get it!

For the context of tw-aggreagator I here propose whatever-is-needed so that the above can be phrased as:

>The tiddlers displayed in the tw-aggregator are not directly displayed. Instead, a static list of available tiddlers is presented on startup [ref #75]. Those tiddlers are just definitions of the real tiddlers, that are stored separately. If you want to see that tiddler just click the title link and you will get it!

...or am I just totally misunderstanding things?

**Further**
I'm thinking that the presented tiddlers in tw-aggreagator are very "discrete" - i.e the data presented from TW1 has nothing to do with the data presented from TW2. The only(?) data that is relevant across the TWs are the tags (right?) and as proposed in #75 these might be a matter for static lists (i.e fast comparad to generated lists).

So, if you open e.g tiddler AndreasHahn then there are a very limited subset of tiddlers relevant. And this subset is predictable, i.e if no tiddlers are loaded from start (merely "the definitions of the real tiddles" as Danielo puts it) then if the uses does open anything concerning AndreasHahn, these tiddlers can all be loaded at once. The "definition" contains a static list of all relevant tiddlers. 

Does this make sense?

**Comment by danielo515** [comment]
Hello @twMat 

My definition is just a personal explanation of what a skinny tiddler is. I abandoned TiddlyDrive because  I got tired of google and because I found couchDB to be a better solution for such task.

**Comment by twMat** [comment]
@danielo515 - thanx. BTW, did you use skinny tiddlers as a way to speed up things or was there another reason?

And, apropos the tw-aggreagator; Do you see anything where aggregators might benefit from your couchDB ideas, or other ideas for that matter? Not that it is "my" project, but I figure it's good to hear your thoughts on the matter because who knows...

**Comment by erwanm** [comment]
@twMat Unfortunately I don't think anything like this kind of "lazy loading" is possible with the CommunitySearch, because for the search to work it is necessary to have all the content loaded in the wiki.

**Comment by danielo515** [comment]
>it is necessary to have all the content loaded in the wiki.
Not necessarily. You can search based on tags and title. Also the definition of how a skinny tiddler should be it's absolutely customizable. You can include a field with keywords, and extend the search to those keywords.

**Comment by erwanm** [comment]
Realizing now that this is related to #73: a light version containing only titles and tags. can be searched on these, and tiddlers links point to the original wiki as usual.

But I wouldn't know how to implement anything more complex than that, so I'm going to give it my new label "unlikely"... sorry! ;)
